x/staking/client/rest/tx.go

```diff
@@ -55,6 +83,92 @@ type (
+//
+// @Summary Generate a create validator transaction
+// @Description Generate a create validator transaction that is ready for signing
+// @Tags staking
+// @Accept  json
+// @Produce  json
+// @Param validatorAddr path string true "The validator address"
+// @Param body body rest.CreateValidatorRequest true "The create validator request payload"
+// @Success 200 {object} rest.createValidator "Returns the unsigned transaction"
+// @Failure 400 {object} rest.ErrorResponse "Returned if the request is invalid"
+// @Failure 401 {object} rest.ErrorResponse "Returned if chain-id required but not present"
+// @Failure 402 {object} rest.ErrorResponse "Returned if fees or gas are invalid"
+// @Failure 500 {object} rest.ErrorResponse "Returned on server error"
+// @Router /staking/validators/{validatorAddr} [post]
+func createValidatorHandlerFn(cliCtx context.CLIContext) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+
+		var req CreateValidatorRequest
+
+		if !rest.ReadRESTReq(w, r, cliCtx.Codec, &req) {
+			return
+		}
+
+		req.BaseReq = req.BaseReq.Sanitize()
+		if !req.BaseReq.ValidateBasic(w) {
+			return
+		}
+
+		pk, err := sdk.GetConsPubKeyBech32(req.Pubkey)
+		if err != nil {
+			rest.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
+			return
+		}
+
+		msg := types.NewMsgCreateValidator(req.ValidatorAddress, pk, req.Value, req.Description, req.Commission, req.MinSelfDelegation)
+		if err := msg.ValidateBasic(); err != nil {
+			rest.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
+			return
+		}
+
+		utils.WriteGenerateStdTxResponse(w, cliCtx, req.BaseReq, []sdk.Msg{msg})
+	}
+}
+
+// editValidatorHandlerFn implements a edit validator handler that is responsible
+// for constructing a properly formatted edit validator transaction for signing.
+//
+// @Summary Generate a edit validator transaction
+// @Description Generate a edit validator transaction that is ready for signing
+// @Tags staking
+// @Accept  json
+// @Produce  json
+// @Param validatorAddr path string true "The validator address"
+// @Param body body rest.EditValidatorRequest true "The edit validator request payload"
+// @Success 200 {object} rest.editValidator "Returns the unsigned transaction"
+// @Failure 400 {object} rest.ErrorResponse "Returned if the request is invalid"
+// @Failure 401 {object} rest.ErrorResponse "Returned if chain-id required but not present"
+// @Failure 402 {object} rest.ErrorResponse "Returned if fees or gas are invalid"
+// @Failure 500 {object} rest.ErrorResponse "Returned on server error"
+// @Router /staking/validators/{validatorAddr} [put]
+func editValidatorHandlerFn(cliCtx context.CLIContext) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+
+		var req EditValidatorRequest
+
+		if !rest.ReadRESTReq(w, r, cliCtx.Codec, &req) {
+			return
+		}
+
+		req.BaseReq = req.BaseReq.Sanitize()
+		if !req.BaseReq.ValidateBasic(w) {
+			return
+		}
+
```

Comment:
The CLI handler of editing a validator passes `nil` for min-self-delegation and commission rate if they were not provided. We need to do the same here.

Reply:
Done.